zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -165,7 +166,11 @@ private boolean doApproval() throws Exception {
             eventLog.compressedZipFiles();
 
             if (Configuration.isUploadNewEndpointEnabled() && DatasetWhitelistChecker.isURIWhitelisted(collectionReader)) {
-                uploadNewEndpoint(collection, collectionReader);
+                List<String> reviewedList = collection.getReviewed().uris().stream().filter(i -> i.contains("datasets")).collect(Collectors.toList());
+                String whitelist = getDatasetWhitelist();
+                List<String> allowed = Arrays.asList(whitelist.split(","));
+                reviewedList = reviewedList.stream().filter(j -> allowed.stream().anyMatch(j::contains)).collect(Collectors.toList());
+                uploadWhitelistedFiles(collectionReader, reviewedList, collection.getDescription().getId());
```

Comment:
We will probably need a null check on the reviewedList to ensure no null pointer exceptions

Reply:
This is now addressed.